packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -418,4 +418,22 @@ class _RenderSingleChildViewport extends RenderBox with RenderObjectWithChildMix
 
     return leadingScrollOffset - (mainAxisExtent - targetMainAxisExtent) * alignment;
   }
+
+  @override
+  void showOnScreen([RenderObject child]) {
+    if (child != null) {
+      // Move viewport the smallest distance to bring [child] on screen.
+      final double leadingEdgeOffset = getOffsetToReveal(child, 0.0);
+      final double trailingEdgeOffset = getOffsetToReveal(child, 1.0);
+      final double currentOffset = offset.pixels;
+      if ((currentOffset - leadingEdgeOffset).abs() < (currentOffset - trailingEdgeOffset).abs()) {
+        offset.jumpTo(leadingEdgeOffset);
+      } else {
+        offset.jumpTo(trailingEdgeOffset);
+      }
+    }
+
+    // Make sure the viewport itself is on screen.
+    super.showOnScreen();
+  }
```

Comment:
The duplication here is unfortunate. If you can't find a clean way to deduplicate, can you add comments in both places that clearly point to the other place so that any bug fixes get applied to both?